### DISCUSSION TITLE
Crash on large script

### DISCUSSION THREAD

**Discussion by martin072** [arendst/Tasmota#13560] [Q&A] [answered]
Could there be an issue with the script console in 10.0? 

Whenever I try to load a script that is bigger than 3584 bytes (3681 bytes), it crashes with an exception.
I am using a Wemos 4Mb and used the  #define USE_UFILESYS

The script with 3584 bytes loads ok, can edit, save etc. but when adding some more code and the script goes up to 3681 bytes, it doesn't save, Tasmota reboots, and I can't load the script editor (only fills half the screen, incomplete page build and some parts of the script visible and sometimes rubbish). Another reboot restores the old script and all is fine again. Could there be an issue with the script editor in the web UI?

I also noticed that Tasmota saves the exception data, is there anyway to do a trace on this (custom compiled .bin with Visual Studio & Platform IO)

**Comment by barbudor** [comment, marked as answer]
See https://github.com/arendst/Tasmota/issues/13465#issuecomment-951597450
You may be over the Script size limit

**Reply by martin072** [reply]
Shows as 8k memory so that would be strange;

`23:23:08.100 Script: nv=18, tv=3, vns=104, vmem=1117, smem=8192 `

**Reply by barbudor** [reply]
As gemu made some changes 9 days ago, please be sure to use the latest dev 10.0.0.X version (not release 10.0.0)

**Comment by gemu2015** [comment]
this is a memory error. i will report this error in console in my next pr.

you may use between 4 to 6 kb of script buffer on an esp8266 when working with the script editor window.
this depends on the free ram of your device e.g.
#define UFSYS_SIZE 4096

if you use the external editor you may use more buffer size (up to 8k) because one internal buffer step is saved.
(see tasmota docs)

**Reply by martin072** [reply]
Isn't this already defined in xdrv_10_scripter? it shows as #define UFSYS_SIZE 8192?

Since the script is still below 4kb, would that still be an issue then? 
Struggling a bit to understand if not using the editor how it works? I see the script is stored as script.txt, but how do I turn off the editor and enable/disable the script if the editor is not used?
 
** Edit **
Sorry, just noticed the link to the external script editor. Will have a look at it.
